clients/src/test/java/org/apache/kafka/clients/producer/internals/ProducerBatchTest.java

```diff
@@ -72,7 +72,11 @@ public void testBatchAbort() throws Exception {
 
         // subsequent completion should be ignored
         batch.done(500L, 2342342341L, null);
-        batch.done(-1, -1, new KafkaException());
+        try {
+            batch.done(-1, -1, new KafkaException());
+        } catch (IllegalStateException e) {
+            assertNotNull(e);
```

Comment:
Is this needed? In what case could e be null?

Reply:
@becketqin restored the test as it was before.